packages/examples/fortune-v2/job-launcher/server/src/modules/job/job.service.ts

```diff
@@ -0,0 +1,254 @@
+import { JobCvatCreateDto, JobFortuneCreateDto, SaveManifestDto, SendWebhookDto } from "./job.dto";
+import { ManifestDto } from "../payment/payment.dto";
+import { EthersSigner, InjectSignerProvider } from "nestjs-ethers";
+import { PaymentSource, PaymentType } from "../../common/enums/payment";
+import { firstValueFrom } from "rxjs";
+import { HttpService } from "@nestjs/axios";
+
+
+@Injectable()
+export class JobService {
+  public readonly logger = new Logger(JobService.name);
+  public readonly storageClient: StorageClient;
+  public readonly storageParams: StorageParams;
+  public readonly bucket: string;
+
+  constructor(
+    @InjectSignerProvider()
+    public readonly ethersSigner: EthersSigner,
+    public readonly jobRepository: JobRepository,
+    public readonly paymentService: PaymentService,
+    public readonly httpService: HttpService,
+    public readonly configService: ConfigService,
+  ) {
+    const storageCredentials: StorageCredentials = {
+      accessKey: this.configService.get<string>("S3_ACCESS_KEY", ""),
+      secretKey: this.configService.get<string>("S3_SECRET_KEY", ""),
+    };
+
+    this.storageParams = {
+      endPoint: this.configService.get<string>("S3_ENDPOINT", "http://127.0.0.1"),
+      port: this.configService.get<number>("S3_PORT", 9000),
+      useSSL: this.configService.get<boolean>("S3_USE_SSL", false),
+    };
+
+    this.bucket = this.configService.get<string>("S3_BUCKET", "launcher");
+
+    this.storageClient = new StorageClient(
+      storageCredentials,
+      this.storageParams
+    );
+  }
+
+  public async createFortuneJob(userId: number, dto: JobFortuneCreateDto): Promise<number> {
+    const { chainId, fortunesRequired, requesterTitle, requesterDescription, price } = dto;
+
+    const userBalance = await this.paymentService.getUserBalance(userId)
+    const amount = BigNumber.from(price).mul(fortunesRequired);
+
+    if (userBalance.lte(amount)) {
+      this.logger.log(ErrorJob.NotEnoughFunds, JobService.name);
+      throw new NotFoundException(ErrorJob.NotEnoughFunds);
+    }
+
+    const manifestData: ManifestDto = {
+      chainId,
+      submissionsRequired: fortunesRequired,
+      requesterTitle,
+      requesterDescription,
+      price,
+      mode: JobMode.DESCRIPTIVE,
+      requestType: JobRequestType.FORTUNE,
+    };
+
+    const { manifestUrl, manifestHash} = await this.saveManifest(manifestData, this.bucket);
+
+    const jobEntity = await this.jobRepository
+      .create({
+        chainId,
+        userId,
+        manifestUrl,
+        manifestHash,
+        status: JobStatus.PENDING,
+        waitUntil: new Date(),
+      });
+
+    if (!jobEntity) {
+      this.logger.log(ErrorJob.NotCreated, JobService.name);
+      throw new NotFoundException(ErrorJob.NotCreated);
+    }
+
+    await this.paymentService.savePayment(userId, PaymentSource.BALANCE, PaymentType.WITHDRAWAL, amount)
```

Comment:
> why is PaymentSource.BALANCE used always?

Since we discussed that all deposits go through the balance, if there are not enough funds on the balance, we add them to the balance, after that we write them off. The balance sheet is a ledger of accounting.